nautobot/core/tests/test_tree_queries.py

```diff
@@ -2,6 +2,25 @@
 from nautobot.dcim.models import Location
 
 
+class TestInvalidateMaxTreeDepthSignal(TestCase):
+    """Tests for the max tree depth cache invalidation signal."""
+
+    def test_invalidate_max_tree_depth_without_tree_fields(self):
+        """Test that max tree depth is not calculated by the invalidate_max_tree_depth signal."""
+        location = Location.objects.first()
+        # Two SAVEPOINT, One SELECT, one UPDATE
+        with self.assertNumQueries(4):
+            Location.objects.without_tree_fields().update_or_create(
```

Comment:
What's the purpose of `update_or_create` here?

Reply:
Normal create should be enough, the original issue was using `update_or_create` which is why I started recreating using that. I will change it.